File: frontend/frontend/src/types/product.ts
// Este archivo declara las interfaces para garantizar la seguridad de tipos en el catálogo de productos.
export interface Product {
    productResourceId: string;
    name: string;
    stock: number;
    price: number;
    imageUrl?: string;
    category: string;
    discountPercentage: number;
    createdAt: string;
}

File: backend/backend/src/InventoryBackend.Domain/Entities/Product.cs
using System;

namespace InventoryBackend.Domain.Entities;

public class Product
{
    public Guid ProductResourceId { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public int Stock { get; set; }
    public decimal Price { get; set; }
    public string ImageUrl { get; set; } = string.Empty;
    public string Category { get; set; } = "General";
}
